csrc/multidevice/executor.cpp

```diff
@@ -98,10 +98,29 @@ MultiDeviceExecutor::MultiDeviceExecutor(
       copyFusionAndChangeOutputs(completeFusion(), vals_to_allocate_);
 }
 
-void MultiDeviceExecutor::postKernel(SegmentedGroup* group) {
+void MultiDeviceExecutor::postKernel(
+    SegmentedGroup* group,
+    const LaunchParams& l_params) {
   if (!should_run_.at(group)) {
     return;
   }
+
+  if (!params_.use_fusion_executor_cache) {
+    // erase from group's input extent corresponding to grid and block dims. Set
+    // them through launchparams.
+    std::vector<Val*> group_input_vals;
+    for (auto input : group->inputs()) {
+      if (input->isA<NamedScalar>()) {
+        NVF_ERROR(input->as<NamedScalar>()->getParallelDim().has_value());
+        NVF_ERROR(l_params.hasDim(
+            input->as<NamedScalar>()->getParallelDim().value()));
+        continue;
```

Comment:
Is this safe to `continue`? A fusion segment may take scalars IIUC.

Reply:
this should also answer this comment
https://github.com/NVIDIA/Fuser/pull/1853#discussion_r1535722505